api/controllers/board.js

```diff
@@ -81,20 +81,45 @@ async function getPublicBoards(req, res) {
 
 async function deleteBoard(req, res) {
   const id = req.swagger.params.id.value;
-  Board.findByIdAndRemove(id, function (err, boards) {
+  const board = Board.findOne({ _id: id }, function (err, board) {
+    if (err) {
+      return res.status(500).json({
+        message: 'Error getting board. ',
+        error: err.message
+      });
+    }
+    if (!board) {
+      return res.status(404).json({
+        message: 'Board does not exist. Board Id: ' + id
+      });
+    }
+    return board;
+  });
+  //  Validate id
+  if (!ObjectId.isValid(id)) {
+    return res.status(404).json({
+      message: 'Invalid ID for a Board. Board Id: ' + id
+    });
+  }
+  if (!req.user.isAdmin && req.user !== board.author) {

```

Comment:
I don't think this works, since `req.user` is an object and `board.author` is a string. The pipeline only runs when we target `master`, so feel free to merge my branch into yours and then change the target branch.

Reply:
Sure. I will give it a deeper look tomorrow. Thanks for the feedback